libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -227,6 +227,20 @@ impl NotificationHandler {
     }
 }
 
+impl Iterator for NotificationHandler {
+    type Item = CmNotification;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.recv.recv() {
+            Ok(notification) => Some(notification),
+            Err(_) => {
+                warn!("connection manager no longer running");
```

Comment:
Need a more accurate error message here, and we shouldn't throw away the root cause.

Reply:
I don't think this is the correct spot to log this as an error condition.  The only error returned is that the channel disconnected.  That could happen due to a clean shutdown or an error condition of the sending end, neither of which can be properly debugged by a message here.  This should be a debug message (as previously suggested), so that we know the receiver has exited, but any error condition should be caught elsewhere.